components/common/src/main/java/com/hotels/styx/common/format/SanitisedHttpHeaderFormatter.java

```diff
@@ -37,8 +38,12 @@
     private final List<String> cookiesToHide;
 
     public SanitisedHttpHeaderFormatter(List<String> headersToHide, List<String> cookiesToHide) {
-        this.headersToHide = requireNonNull(headersToHide);
-        this.cookiesToHide = requireNonNull(cookiesToHide);
+        this.headersToHide = Collections.unmodifiableList(requireNonNull(headersToHide));
```

Comment:
This doesn't stop the passed lists from being modified it just wraps them in a facade that stops modification methods. To protect this object's state you will need to copy the list contents.